### ISSUE TITLE
Training time

### ISSUE THREAD

**Issue by xingzhejun** [dllm-reasoning/d1#25]
Hi!

A magnificant work! A lot of insights which motivate me in terms of dllm, thank you!

I am now trying to reproduce the diff-grpo of gsm8k, and I have noticed you set num_train_epochs=10. However, when I ran the code, I found it would take around 4days to finish them all even with 8 H100/ B200 Gpus. May I ask around how many iterations you can get the approxiamte results in the paper? Or there is something I may make a mistake?

Another issue is that since many morks like fast-dllm have proposed the kv-cache methods, have you tried incorperating them in this work? Or is there any other works that already did?

Thank you!

**Comment by siyan-zhao** [comment]
Hi, Yes the total training time is huge because we set train epoch to 10, but we don’t need it to complete all epochs. It can converge earlier. You can early stop the experiments if the reward curve converges. for us it take ~1-2 days to converge. We train 7700, 6600 steps (number of gradient updates) for GSM8K and MATH500 respectively; for Countdown and Sudoku,
we train for 5000, 3800 steps respectively.

I tried fast-dllm in online RL sampling but it did not work very stably. The logits became more high and overconfident during training, which caused it to generate many tokens at once and degrade performance. i think it is an interesting direction to explore how to make it stable.
